File: ru/2/1/Homework1/Hw1/Calculator.cs
namespace Hw1;

public static class Calculator
{
    public static double Calculate(double value1, CalculatorOperation operation, double value2)
    {
        return operation switch
        {
            CalculatorOperation.Plus => value1 + value2,
            CalculatorOperation.Minus => value1 - value2,
            CalculatorOperation.Multiply => value1 * value2,
            CalculatorOperation.Divide => value1 / value2,
            _ => throw new ArgumentOutOfRangeException()
        };
    }
}

File: ru/2/1/Homework4/Hw4/Calculator.fs
module Hw4.Calculator

open System

type CalculatorOperation =
     | Plus = 0
     | Minus = 1
     | Multiply = 2
     | Divide = 3
     | Undefined = 4
     
let calculate (value1 : float) (operation : CalculatorOperation) (value2 : float) =
    NotImplementedException() |> raise
    
